### PR TITLE
attempt to fix binding of socket

### PR DESCRIPTION
**Pull request by starkillerOG**
This is an attempt to resolve this issue: https://github.com/home-assistant/core/issues/37306

### CODE REVIEW

File: xiaomi_gateway/__init__.py [hunk 1]
```diff
@@ -194,7 +191,7 @@ class XiaomiGateway:
     """Xiaomi Gateway Component"""
 
     # pylint: disable=too-many-arguments
-    def __init__(self, ip_adress, port, sid, key, discovery_retries, interface, proto=None):
+    def __init__(self, ip_adress, port=MULTICAST_PORT, sid, key, discovery_retries, interface, proto=None):
```
Comment (fliphess) [inline xiaomi_gateway/__init__.py]: You can't do named arguments before unnamed arguments, so this has to become something like:
```
    def __init__(self, ip_adress, sid, key, discovery_retries, interface, port=MULTICAST_PORT, proto=None):
        ....
```

**Comment by dshokouhi** [comment]
All the changes here and mentioned in the ticket (https://github.com/home-assistant/core/issues/37306#issuecomment-653622458) have also resolved #171 for me

**Comment by starkillerOG** [comment]
Tested it and it still works for me.
Although it has always worked on my setup....

@fliphess can you test to see if it also fixes your issue....

**Comment by meichthys** [comment]
@dshokouhi Could you outline how you tested this? Did you modify the existing integration in your instance, or did you setup a custom integration with the above changes?

If you modified your existing integration, could you let me know where you found the __init__.py file to modify? I can't seem to locate it on a HASOS install.

**Comment by dshokouhi** [comment]
@meichthys as I run my dev environment in a venv I am able to modify the system files directly.   In my venv this file is located at:

```
/srv/homeassistant/lib/python3.7/site-packages/xiaomi_gateway/__init__.py
```

You can run the changes for Home Assistant in a custom component by copying all the files from the branch mentioned in the comment https://github.com/home-assistant/core/issues/37306#issuecomment-653622458.  There is a lot going on here with this change because we need parent library updates from this PR as well as HA changes mentioned in the comment.

You'll need to find out where yours is located for your install, if you are on the HA discord you can just ask in the installation channel to get quick help.

**Comment by meichthys** [comment]
@dshokouhi Ok, Thanks. I asked in Discord and it sounds like it's recommended to just setup a custom component. 
I'll give it a shot and report back here with my results.

**Comment by meichthys** [comment]
I get this after setting up the custom component and rebooting:
image
Any ideas?

**Comment by dshokouhi** [comment]
@meichthys thats what i was trying to explain to you.  You can't just run a custom component here, you must include the parent library changes which requires either this PR to be merged a new release to be pubished to pypi or you need to modify the file directly.  Custom components are only for the HA side of things.

**Comment by meichthys** [comment]
I see. I'll be patient and wait for the next release.

**Comment by borland502** [comment]
I'll be a good boy too then.  I can set up only one of the gateways through the flow process, where as prior to 114 both worked in yaml config

**Comment by meichthys** [comment]
For me I can’t even get a single instance of the integration working, although most others seem to only have issues with multiple instances.

**Comment by starkillerOG** [comment]
@syssi @Danielhiversen could you merge this?
I have tested it on my enviroment and it actually also improves it for users not experiencing problems:
- it ensures proper closure of the xiaomi_aqara integration while it used to hang for me.
- now I can discover the gateway using interface "any", It used to only work when I set interface to the IP-adress of my HomeAssistant device.

It schould also fix some issues for users experincing binding problems of the socket.

**Comment by borland502** [comment]
I'm not sure this is useful, but as I said earlier...  One gateway works, one does not now.  However, even on another home assistant I cannot bind to the gateway that would not be discovered.  I know I have the same generation, but different v3 gateway models because I had to perform surgery to open the software port.  Does this potential problem have an issue even with different home assistants?

**Comment by borland502** [comment]
Success.  I updated the firmware on the non-responsive one to 1.4.1_176.0220 and was able to successfully configure both.  The one that worked before is still on an older firmware.

**Comment by starkillerOG** [comment]
Thanks for merging!

**Pull request merged**